iocore/net/SSLNetVConnection.cc

```diff
@@ -1099,7 +1099,7 @@ SSLNetVConnection::sslServerHandShakeEvent(int &err)
   if (sslHandshakeHookState == HANDSHAKE_HOOKS_PRE) {
     if (!curHook) {
       Debug("ssl", "Initialize preaccept curHook from NULL");
-      curHook = ssl_hooks->get(TS_VCONN_START_INTERNAL_HOOK);
+      curHook = ssl_hooks->get(TSSslHookInternalID(TS_VCONN_START_HOOK));
```

Comment:
Are these explicit conversions required? I would think since the constructor is not marked `explicit` it should be possible to just pass the `TSHttpHookID`.

Reply:
They may not be necessary.  I think it might be confusing to maintainers to omit the explicit conversion.  I think this is Leif-legal since I'm calling an actual constructor, not using a constructor style cast.